common/src/main/java/org/opensearch/ml/common/parameter/AnomalyDetectionParams.java

```diff
@@ -0,0 +1,176 @@
+        Double cost = null;
+        Double coeff = null;
+        Double epsilon = null;
+        Integer degree = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case KERNEL_FIELD:
+                    kernelType = ADKernelType.valueOf(parser.text().toUpperCase(Locale.ROOT));
+                    break;
+                case GAMMA_FIELD:
+                    gamma = parser.doubleValue();
+                    break;
+                case NU_FIELD:
+                    nu = parser.doubleValue();
+                    break;
+                case COST_FIELD:
+                    cost = parser.doubleValue();
+                    break;
+                case COEFF_FIELD:
+                    coeff = parser.doubleValue();
+                    break;
+                case EPSILON_FIELD:
+                    epsilon = parser.doubleValue();
+                    break;
+                case DEGREE_FIELD:
+                    degree = parser.intValue();
+                    break;
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }
+        return new AnomalyDetectionParams(kernelType, gamma, nu, cost, coeff, epsilon, degree);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return PARSE_FIELD_NAME;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (kernelType == null) {
+            out.writeBoolean(false);
+        } else {
+            out.writeBoolean(true);
+            out.writeEnum(kernelType);
+        }
+        out.writeOptionalDouble(gamma);
+        out.writeOptionalDouble(nu);
+        out.writeOptionalDouble(cost);
+        out.writeOptionalDouble(coeff);
+        out.writeOptionalDouble(epsilon);
+        out.writeOptionalInt(degree);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (kernelType != null) {
+            builder.field(KERNEL_FIELD, kernelType);
+        }
+        if (gamma != null) {
+            builder.field(GAMMA_FIELD, gamma);
+        }
+        if (nu != null) {
+            builder.field(NU_FIELD, nu);
+        }
+        if (cost != null) {
+            builder.field(COST_FIELD, cost);
+        }
+        if (coeff != null) {
+            builder.field(COEFF_FIELD, coeff);
+        }
+        if (epsilon != null) {
+            builder.field(EPSILON_FIELD, epsilon);
+        }
+        if (degree != null) {
+            builder.field(DEGREE_FIELD, degree);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public int getVersion() {
```

Comment:
Just curious: this version is always 1?

Reply:
We set version as 1 for first release. May change to higher version in future